### DISCUSSION TITLE
Store an array in a variable and loop over it: `MY_VAR := ["hello", "world"]`

### DISCUSSION THREAD

**Discussion by 0xbuidler** [casey/just#2429] [Q&A]
I would like to store an array in a variable such as `MY_VAR := ["hello", "world"]` and loop over it in my recipe. How can I do that ? Is it even possible ?

**Comment by laniakea64** [comment]
This is not possible.  All `just` variables are strings.

Would something along the lines of https://github.com/casey/just/issues/2370 work for your case?  i.e. storing the variable as a string with elements separated by a "delimiter string" that is not present in any element, then having the recipe's shell/interpreter split on that "delimiter string" into its own array type and loop over the result?

**Comment by goyalyashpal** [comment]
what is the use case scenario?  

also see [the XY problem](https://xyproblem.info/ "Home - The XY Problem")
